packages/esm-outpatient-app/src/past-visit/past-visit-details/past-visit-summary.component.tsx

```diff
@@ -64,9 +78,76 @@ const PastVisitSummary: React.FC<PastVisitSummaryProps> = ({ encounters, patient
             });
           }
         });
+
+        // Iterating through every Encounter
+        encounters.forEach((enc: Encounter) => {
+          // Check for Visit Diagnoses and Notes
+          if (enc.encounterType.display === 'Visit Note') {
+            enc.obs.forEach((obs: Observation) => {
+              if (obs.concept && obs.concept.display === 'Visit Diagnoses') {
+                // // Putting all the diagnoses in a single array.
+                diagnoses.push({
+                  diagnosis: obs.groupMembers.find((mem) => mem.concept.display === 'PROBLEM LIST').value.display,
+                });
+              } else if (obs.concept && obs.concept.display === 'General patient note') {
+                // Putting all notes in a single array.
+                notes.push({
+                  note: obs.value,
+                  provider: {
+                    name: enc.encounterProviders.length ? enc.encounterProviders[0].provider.person.display : '',
+                    role: enc.encounterProviders.length ? enc.encounterProviders[0].encounterRole.display : '',
+                  },
+                  time: formatTime(parseDate(obs.obsDatetime)),
+                  concept: obs.concept,
+                });
+              }
+            });
+          }
+        });
+      });
+
+      encounter?.obs.forEach((obs: Observation) => {
+        if (obs.concept?.uuid === config.concepts.pulseUuid) {
+          vitals.push({
+            pulse: obs.value,
+            provider: {
+              name: encounter.encounterProviders.length ? encounter.encounterProviders[0].provider.person.display : '',
+              role: encounter.encounterProviders.length ? encounter.encounterProviders[0].encounterRole.display : '',
+            },
+            time: formatTime(parseDate(obs.obsDatetime)),
+          });
+        } else if (obs.concept?.uuid === config.concepts.oxygenSaturationUuid) {
+          vitals.push({
+            oxygenSaturation: obs.value,
+          });
+        } else if (obs.concept?.uuid === config.concepts.respiratoryRateUuid) {
+          vitals.push({
+            respiratoryRate: obs.value,
+          });
+        } else if (obs.concept?.uuid === config.concepts.temperatureUuid) {
+          vitals.push({
+            temperature: obs.value,
+          });
+        } else if (obs.concept?.uuid === config.concepts.systolicBloodPressureUuid) {
+          vitals.push({
+            systolic: obs.value,
+          });
+        } else if (obs.concept?.uuid === config.concepts.diastolicBloodPressureUuid) {
```

Comment:
This section already exists, can you just refactor and re-use accordingly. Thanks